sdk/core/azure_core/src/policies/storage_bearer_token_policy.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{
+    auth::TokenCredential,
+    policies::{Policy, PolicyResult},
+    Context, Request,
+};
+use async_trait::async_trait;
+use std::sync::Arc;
+use tracing::debug;
+
+#[derive(Debug, Clone)]
+pub struct BearerTokenCredentialPolicy<'a> {
+    credential: Arc<dyn TokenCredential>,
+    scopes: &'a [&'a str],
+}
+
+impl<'a> BearerTokenCredentialPolicy<'a> {
+    pub fn new(credential: Arc<dyn TokenCredential>, scopes: &'a [&'a str]) -> Self {
+        Self { credential, scopes }
+    }
```

Comment:
If you stored a `Vec<String>`, you'd want to take a slice of `Into<String>`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=548f8e639f4e9216eec24c5d53f83923

Reply:
I can take a shot at this implementation. Thanks for the helpful playground link, what a nifty map!